src/collective/behavior/seo/browser/structured_data.py

```diff
@@ -0,0 +1,114 @@
+from plone.app.layout.viewlets.common import ViewletBase
+from plone.event.interfaces import IEventAccessor
+from plone.app.uuid.utils import uuidToObject
+from plone import api
+import json
+import pkg_resources
+
+
+try:
+    pkg_resources.get_distribution('collective.address')
+except pkg_resources.DistributionNotFound:
+    HAS_VENUE = False
+else:
+    from collective.venue.interfaces import IVenueEnabled
+    from collective.address.vocabulary import get_pycountry_name
+    HAS_VENUE = True
+
```

Comment:
@mauritsvanrees and me reviewed this. We ask ourselves what the next developer should do who want to add custom json-ld to this viewlet.  Add more try except imports here? 

The generic solution would be to have a named adapter lookup on portal type and let the add'ons  (collective.venue, collective.address or your projects policy package) have adapters for the customer CT's Then the adapters are queried and called by this viewlet so that the data is enriched but included in one viewlet).   A bit like the body css class adapter system.

This is maybe overkill for the first 'custom addon' support, but more extendable and cleaner.

Reply:
my idea is to be able to add a behavior based on the json-ls-formats that are available, so you could add the format that fits a default or your custom type best.